File: ios/AwesomeLibrary.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AwesomeLibrary, NSObject)

// Export a native method called 'multiply' to JavaScript Library
RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

// Export a native method called 'openCamera' to JavaScript Library
RCT_EXTERN_METHOD(openCamera:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/AwesomeLibrary.swift
@objc(AwesomeLibrary)
class AwesomeLibrary: NSObject {

  @objc(multiply:withB:withResolver:withRejecter:)
  func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(a*b)
  }

  @objc(openCamera:withRejecter:)
  func openCamera(resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
    let imagePicker = UIImagePickerController()
    imagePicker.sourceType = .photoLibrary;

    DispatchQueue.main.async {
      RCTPresentedViewController()?.present(imagePicker, animated: true, completion: nil)
    }

    resolve("Camera iOS aberta com sucesso!");
  }
}
